mono/metadata/unity-liveness.c

```diff
@@ -600,75 +503,35 @@ LivenessState* mono_unity_liveness_allocate_struct (MonoClass* filter, guint max
 	// if all_objects run out of space, run through list, add objects that match the filter, clear bit in vtable and then clear the array.
 
 	state = g_new0(LivenessState, 1);
-	max_count = max_count < 1000 ? 1000 : max_count;
-	state->all_objects = array_create_and_initialize(max_count*4);
-	state->process_array = array_create_and_initialize (max_count);
 
-	state->first_index_in_all_objects = 0; 
 	state->filter = filter;
 	state->traverse_depth = 0;
 
 	state->callback_userdata = callback_userdata;
 	state->filter_callback = callback;
-	state->onWorldStartCallback = onWorldStartCallback;
-	state->onWorldStopCallback = onWorldStopCallback;
+	state->reallocateArray = reallocateArray;
+
+	state->all_objects = block_array_create(state);
+	state->process_array = block_array_create(state);
 
 	return state;
 }
 
-void mono_unity_liveness_finalize (LivenessState* state)
+void mono_unity_liveness_finalize(LivenessState *state)
 {
-	int i;
-	for (i = 0; i < state->all_objects->len; i++)
-	{
-		MonoObject* object = g_ptr_array_index(state->all_objects,i);
+	block_array_reset_iterator(state->all_objects);
+	gpointer it = block_array_next(state->all_objects);
+	while (it != NULL) {
+		MonoObject *object = it;
 		CLEAR_OBJ(object);
+		it = block_array_next(state->all_objects);
 	}
 }
 
-void mono_unity_liveness_free_struct (LivenessState* state)
+void mono_unity_liveness_free_struct(LivenessState *state)
 {
 	//cleanup the liveness_state
-	array_destroy(state->all_objects);
-	array_destroy(state->process_array);
+	block_array_destroy(state->all_objects, state);
+	block_array_destroy(state->process_array, state);
 	g_free(state);
 }
-
-void mono_unity_liveness_stop_gc_world (LivenessState* state)
```

Comment:
Where did this go? How do we not stop the world anymore?

Reply:
using generic scriptin_gc_stop/start_world now as callback is not needed